editor/src/components/canvas/ui-jsx-canvas.tsx

```diff
@@ -101,9 +101,43 @@ export const ElementsToRerenderGLOBAL: { current: ElementsToRerender } = {
   current: 'rerender-all-elements',
 }
 
+export type FileRootPath = {
+  type: 'file-root'
+}
+
+export function fileRootPathToString(fileRootPath: FileRootPath): string {
+  return 'file-root'
+}
+
+export function fileRootPathFromString(fileRootPathString: string): FileRootPath {
+  if (fileRootPathString === 'file-root') {
+    return { type: 'file-root' }
+  } else {
+    throw new Error(`Unknown file root path string: ${fileRootPathString}`)
+  }
+}
+
+export function insertionCeilingToString(insertionCeiling: ElementPath | FileRootPath): string {
```

Comment:
this is not used, is it? 🤔

Reply:
it is used in `scoped-variables.spec.tsx`

Reply:
oh apologies, I misclicked - I meant `insertionCeilingFromString`